src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/dllmain.cpp

```diff
@@ -199,7 +199,7 @@ HRESULT WINAPI RoActivateInstanceDetour(HSTRING activatableClassId, IInspectable
     // Activate in current apartment
     if (location == ActivationLocation::CurrentApartment)
     {
-        IActivationFactory* pFactory;
+        Microsoft::WRL::ComPtr<IActivationFactory> pFactory;
```

Comment:
is there a reason you are not using `winrt::com_ptr`?
consider removing the `p` prefix on the members that indicate raw pointer.

Reply:
This code doesn't have a WIL dependency (it has a local definition for RETURN_IF_FAILED). It probably should be updated to use WIL, but that's a larger change not related to the bug fix at hand. I've filed a new issue: https://github.com/microsoft/xlang/issues/650